backend.go

```diff
@@ -11,6 +11,38 @@ import (
 	"github.com/hashicorp/vault/sdk/queue"
 )
 
+type backend struct {
+	*framework.Backend
+	sync.RWMutex
+	client ldapClient
+
+	// CredRotationQueue is an in-memory priority queue used to track Static Roles
+	// that require periodic rotation. Backends will have a PriorityQueue
+	// initialized on setup, but only backends that are mounted by a primary
+	// server or mounted as a local mount will perform the rotations.
+	//
+	// cancelQueue is used to remove the priority queue and terminate the
+	// background ticker.
+	credRotationQueue *queue.PriorityQueue
+	cancelQueue       context.CancelFunc
+
+	// managedUsers contains the set of LDAP usernames managed by the secrets engine
+	// static role and check-in/check-out systems. It is used to ensure that users
+	// are exclusively managed by one system and not both. Access to managedUsers is
+	// synchronized by the managedUserLock.
+	managedUsers    map[string]struct{}
+	managedUserLock sync.Mutex
```

Comment:
This is new code introduced by this PR. See comment for its purpose.

Reply:
This is one approach at exclusively managing users in library sets and static roles. A different approach that I thought of involved listing all static roles and library sets from Vault storage every time we manage a new user. It's simpler but has different performance tradeoffs. Happy to discuss.